spaces/importers/tprek_unit_importer.py

```diff
@@ -55,3 +63,57 @@ def _update_unit(self, unit: Unit, tprek_unit_data: TprekUnitData, tprek_locatio
         Location.objects.update_or_create(unit=unit, defaults=asdict(tprek_location_data))
 
         self.updated_units_count += 1
+        if unit.origin_hauki_resource is None:
+            self.units_for_hauki_import.append(unit)
+
+
+class TprekUnitHaukiResourceIdImporter:
+    @classmethod
+    @atomic
+    def import_hauki_resources_for_units(cls, units: list[Unit]) -> None:
+        hauki_resource_id_map = cls._fetch_hauki_resource_ids(units)
+
+        logger.info(f"Importing Hauki resources for {len(units)} units...")
+
+        updated_units_count = 0
+        created_resources_count = 0
+        for unit in units:
+            hauki_resource_id = hauki_resource_id_map.get(str(unit.tprek_id))
+            if hauki_resource_id:
+                origin_hauki_resource, created = OriginHaukiResource.objects.get_or_create(id=hauki_resource_id)
+                unit.origin_hauki_resource = origin_hauki_resource
+                unit.save()
```

Comment:
Should the atomic be here? If so, use it here as a context manager so that it's clear which operations need to be atomic.